src/TSTransformer/nodes/transformStatementList.ts

```diff
@@ -66,10 +77,9 @@ export function transformStatementList(
 		}
 	}
 
-	if (state.compilerOptions.removeComments !== true && statements.length > 0) {
-		const lastStatement = statements[statements.length - 1];
-		const lastToken = lastStatement.parent.getLastToken();
-		if (lastToken && !ts.isNodeDescendantOf(lastToken, lastStatement)) {
+	if (state.compilerOptions.removeComments !== true) {
+		const lastToken = getLastToken(statements) ?? endOfFileToken;
```

Comment:
Should we not do both of these?
I'm thinking of the case eg
```ts
// A
print("cool")
// B
;
// C
```
Would ; classify as last token and have the B comment, while end of file gets the C comment? We'd be missing out on one by doing the null-coalescing here.

Reply:
`// B` is a mid-statement comment which we don't really support right now.. maybe we eventually should

I'll leave a comment to better explain this better, but:
- `getLastToken(statements)` will only return `undefined` if `statements` is empty.
- This can happen in the case of a SourceFile with only comments, in that case we have to pass the endOfFileToken.

Reply:
In the case of a SourceFile with non-empty statements list, `getLastToken(statements)` should always be equal to `endOfFileToken`. If we did both, we'd duplicate the final comments.

Reply:
> `// B` is a mid-statement comment which we don't really support right now.. maybe we eventually should

Would it not technically be a separate-statement comment on a `ts.EmptyStatement`? But I presume that's also how the comment would still be emitted then.
Either way, with the token explanation, my first reaction is null and void.